### ISSUE TITLE
`<URL>` cannot be parsed

### ISSUE THREAD

**Issue by pgundlach** [speedata/publisher#529]
### Discussed in https://github.com/speedata/publisher/discussions/528

<div type='discussions-op-text'>

<sup>Originally posted by **pr-apes** October 18, 2023</sup>
@pgundlach 

I have the following layout (`testing.xml`):

``` xml
<Layout xmlns="urn:speedata.de:2009/publisher/en"
  xmlns:sd="urn:speedata:2009/publisher/functions/en">

  <Record element="datos">
    <PlaceObject>
      <Textblock>
        <Paragraph>
            <A href="{fila/c_6}">
              <URL><Value select="fila/c_6" /> </URL>
            </A>
        </Paragraph>
      </Textblock>
    </PlaceObject>
  </Record>
</Layout>
```

With the following `data.xml`:

``` xml
<datos><fila><c_6>https://a.b/c</c_6></fila></datos>
```

I get fine validation from both:

```
xmllint --noout testing.xml && xmllint --noout --relaxng ../speedata-publisher/share/schema/layoutschema-en.rng testing.xml
```

But `sp --layout testing.xml` gives the following error:

````
[…]\commands.lua:4633: invalid value (table) at index 1 in table for 'concat'
stack traceback:
        ...tilidades\speedata-publisher\sw/lua/publisher\spinit.lua:54: in function 'call'
        ...tilidades\speedata-publisher\sw/lua/publisher\spinit.lua:397: in function 'main_loop'
        ...tilidades\speedata-publisher\sw/lua/publisher\spinit.lua:406: in main chunk
        [C]: in function 'require'
        [\directlua]:1: in main chunk
Stop processing data
1 errors occurred
````

From reasons unknown to me, `<URL>` cannot be parsed. It has to be my stupid mistake (but I ignore which one 😅).

Many thanks for your help.</div>

**Comment by pgundlach** [comment]
This is a bug. Will fix. Thank you for the report. I don't have a workaround yet.

**Comment by pr-apes** [comment]
Many thanks for your fast reply, @pgundlach.

It drove me crazy, since I didn't understand what was wrong in my code.

**Comment by pgundlach** [comment]
Not sure if this fix (in version 4.15.6) really fixes it. It is a bit of a rewrite.

**Comment by pr-apes** [comment]
Many thanks for the fix, @pgundlach.

I have another layout to test this with, so I'll report later.
